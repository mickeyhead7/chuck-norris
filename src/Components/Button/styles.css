.button {
  background-color: green;
  border: 0;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1rem;
  padding: 1rem;
}
