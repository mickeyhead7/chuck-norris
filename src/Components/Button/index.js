import React from 'react';
import './styles.css';

const Button = ({ disabled, onClick }) => (
  <button
    className="button"
    disabled={disabled}
    onClick={onClick}
  >
    Get a new joke
  </button>
);

export default Button;
