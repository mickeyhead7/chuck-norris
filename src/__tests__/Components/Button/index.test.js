import React from 'react';
import { mount, shallow } from 'enzyme';
import Button from '../../../Components/Button';

describe('Button', () => {
  test('Button default', () => {
    const buttonClicked = jest.fn();
    const props = {
      disabled: false,
      onClick: buttonClicked
    };
    const button = shallow(<Button {...props} />);
    
    expect(button).toMatchSnapshot();
    expect(button.props().disabled).toBeFalsy();
    
    button.simulate('click');
    
    expect(buttonClicked).toHaveBeenCalledTimes(1);
  });

  test('Button disabled', () => {
    const buttonClicked = jest.fn();
    const props = {
      disabled: true,
      onClick: buttonClicked
    };
    const button = mount(<Button {...props} />);
    
    expect(button).toMatchSnapshot();
    expect(button.props().disabled).toBeTruthy();
    
    button.simulate('click');
    
    expect(buttonClicked).toHaveBeenCalledTimes(0);
  });
});