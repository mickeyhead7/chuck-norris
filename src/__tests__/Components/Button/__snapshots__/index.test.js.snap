// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={false}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Get a new joke",
      "className": "button",
      "disabled": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Get a new joke",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Get a new joke",
        "className": "button",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Get a new joke",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button Button disabled 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Button
    disabled={true}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "disabled": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <button
        class="button"
        disabled=""
>
        Get a new joke
</button>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Get a new joke",
        "className": "button",
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Get a new joke",
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="button"
          disabled=""
>
          Get a new joke
</button>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get a new joke",
          "className": "button",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Get a new joke",
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
